libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/FeatureProcessingTask.java

```diff
@@ -14,39 +14,57 @@
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.mapbox.services.android.navigation.ui.v5.route.RouteConstants.PRIMARY_ROUTE_PROPERTY_KEY;
 
-class FeatureProcessingTask extends AsyncTask<Void, Void, Void> {
+class FeatureProcessingTask extends Thread {
 
   private final List<DirectionsRoute> routes;
   private final List<FeatureCollection> routeFeatureCollections = new ArrayList<>();
   private final WeakReference<OnRouteFeaturesProcessedCallback> callbackWeakReference;
   private final HashMap<LineString, DirectionsRoute> routeLineStrings = new HashMap<>();
+  private AtomicBoolean cancelThread = new AtomicBoolean(false);
+  private Handler postHandler;
 
-  FeatureProcessingTask(List<DirectionsRoute> routes, OnRouteFeaturesProcessedCallback callback) {
+  FeatureProcessingTask(List<DirectionsRoute> routes, OnRouteFeaturesProcessedCallback callback, Handler handler) {
```

Comment:
looks like we're passing ref to the list here, shouldn't we do a deep copy of it?